### PR TITLE
Enemy tests

### PR DESCRIPTION
**Pull request by lucydotc**
Added tests for:

all getter methods
prepEnemy()
receiveStatus()
enemyShootAt()

### CODE REVIEW

File: src/main/java/seedu/address/model/player/Enemy.java [hunk 1]
```diff
@@ -395,15 +394,15 @@ private void modeCleanup(Coordinates usedCoord) {
      * checks that coord is made up of one odd and one even x-y coordinate pair
      * and returns true if so
      */
-    private boolean hasParity (int row, int col) {
+    public boolean hasParity (int row, int col) {
```
Comment (iwle) [inline src/main/java/seedu/address/model/player/Enemy.java:395]: The access modifier is changed here because of testing purposes?

Comment (iwle) [reply src/main/java/seedu/address/model/player/Enemy.java:395]: If it is, then nevermind. Else the method can be private.

File: src/main/java/seedu/address/model/player/Enemy.java [hunk 2]
```diff
@@ -395,15 +394,15 @@ private void modeCleanup(Coordinates usedCoord) {
...
 
         int parity = (row % 2) + (col % 2);
         return (parity == 1);
     }
 
     /*************************************************************
      *  puts everything back from the stack to the drawing bags
-     *  Idea here is that once something is destroyed,
+     *  Idea here is that once something is destroyed,1
```
Comment (iwle) [inline src/main/java/seedu/address/model/player/Enemy.java:403]: There’s additional two letters at the end of this sentence.

File: src/test/java/seedu/address/model/player/EnemyTest.java [hunk 3]
```diff
@@ -1,95 +1,214 @@
...
-    public final Enemy testPlayer = new Enemy();
+    private Enemy testEnemy;
+    private Player testPlayer;
+    private BattleManager batman;
 
-    //Lucy TO-FIX:
-    /*  @Test
+    @Before
+    public void readyEnemyForTesting() {
+
```
Comment (iwle) [inline src/test/java/seedu/address/model/player/EnemyTest.java:31]: Minor problem: can remove the newline here.

File: src/test/java/seedu/address/model/player/EnemyTest.java [hunk 4]
```diff
@@ -1,95 +1,214 @@
...
+        return new HashSet<>(list1).equals(new HashSet<>(list2));
+    }
 
+    //Lucy TO-FIX:
+    @Test
     public void constructor_default() {
-        Player newPlayer1 = new Player();
-        Player newPlayer2 = new Player("Player1", 5);
-        assertEquals(newPlayer1, newPlayer2);
+        //stub
```
Comment (iwle) [inline src/test/java/seedu/address/model/player/EnemyTest.java]: There is no need to stub this portion, the test cases here can be used to unit test the equals() method.

**Review by bos10 — APPROVED** [review]
Logging consistently within your code makes every step accountable and debugging is made alot easier. The addition of these test cases and its successful passing has contributed to making our application more robust and bug-free! Great job Lucy

File: src/test/java/seedu/address/model/player/EnemyTest.java [hunk 5]
```diff
@@ -1,95 +1,214 @@
...
+    /**
+     * Expected result:
+     *    For every call of enemyShootAt:
+     *    targetHistory must + 1
+     *    allTargets must - 1
+     *    if watchlist empty -> allParityTargets must - 1
+     *    else watchlist must - 1
+     */
+    @Test public void test_enemyShootAt() {
+
```
Comment (iwle) [inline src/test/java/seedu/address/model/player/EnemyTest.java:167]: Minor problem: can remove the newline here.

File: src/test/java/seedu/address/model/player/EnemyTest.java [hunk 6]
```diff
@@ -1,95 +1,214 @@
...
+        int startingAllParityTargets;
+        int startingWatchlistSize;
+        int startingTargetHistorySize;
+
+        setUpAllShips(testPlayer);
+        batman = new BattleManager(testPlayer, testEnemy);
+
+        for (int i = 0; i < 5; i++) {
+
+            startingAllPossibleTargetsSize = testEnemy.getAllPossibleTargets().size();
```
Comment (iwle) [inline src/test/java/seedu/address/model/player/EnemyTest.java:177]: Newline can remove.

File: src/main/java/seedu/address/model/player/Enemy.java [hunk 7]
```diff
@@ -417,4 +416,22 @@ private void cleanseWatchlist () {
 
         }
     }
+
+    public Status getLastAttackStatus() {
+        return this.lastAttackStatus;
+    }
+    public ArrayList<Coordinates> getAllPossibleTargets() {
```
Comment (iwle) [inline src/main/java/seedu/address/model/player/Enemy.java:421]: Need to add a newline here to separate from previous method.

File: src/main/java/seedu/address/model/player/Enemy.java (hunk 1 above)
Comment (s-tr) [reply src/main/java/seedu/address/model/player/Enemy.java:395]: If you want to test this, it is better to have it be package-private (no access modifier)

**Review by iwle — APPROVED** [review]
Looks good to go.

**Pull request merged**